task/thermal/src/control.rs

```diff
@@ -918,31 +918,36 @@ impl<'a> ThermalControl<'a> {
         // they are, so someone else has to do that.
     }
 
-    /// Returns an iterator over tuples of `(value, thermal model)`
+    /// Returns an iterator over tuples of `(sensor_id, value, thermal model)`
     ///
-    /// The `values` array must contain `static_inputs.len()` +
-    /// `dynamic_inputs.len()` values, in that order; this function will panic
-    /// otherwise.
+    /// The `values` array contains static and dynamic values (in order);
+    /// this function will panic if sizes are mismatched.
+    ///
+    /// Every dynamic input is represented by an `Option<DynamicInputChannel>`.
+    /// If the input is not present right now, it will be `None`, but will
+    /// continue to take up space to preserve ordering.
     ///
     /// In cases where dynamic inputs are not present (i.e. they are `None` in
     /// the array), the iterator will skip that entire tuple.
     fn zip_temperatures<'b, T>(
-        values: &'b [T],
-        (static_inputs, dynamic_inputs): (
-            &'b [InputChannel],
-            &'b [Option<DynamicInputChannel>],
-        ),
-    ) -> impl Iterator<Item = (&'b T, ThermalProperties)> {
-        assert_eq!(values.len(), static_inputs.len() + dynamic_inputs.len());
-        values
+        bsp: &'b Bsp,
+        values: &'b [T; TEMPERATURE_ARRAY_SIZE],
+        dynamic_channels: &'b [Option<DynamicInputChannel>;
+                bsp::NUM_DYNAMIC_TEMPERATURE_INPUTS],
```

Comment:
nit, take it or leave it: maybe consider a 
```rust
type DynamicChannelsArray = [
    Option<DynamicInputChannel>;
    bsp::NUM_DYNAMIC_TEMPERATURE_INPUTS
];
```

or something? Normally, I probably wouldn't suggest this, but the weird rustfmt wrapping here is honestly kind of hard to interpret at first glance...